static/js/publisher/release/contextualMenu.js

```diff
@@ -44,37 +44,36 @@ export default class ContextualMenu extends Component {
   }
 
   renderIcon() {
-    return <i className="p-icon--contextual-menu" />;
-  }
-
-  renderItems() {
-    return null;
+    return this.props.icon || <i className="p-icon--contextual-menu" />;
   }
 
   render() {
-    const { position } = this.props;
+    const { isDisabled, position } = this.props;
     const menuClass = "p-contextual-menu" + (position ? `--${position}` : "");
     const appearance = this.props.appearance || "neutral";
-    const className = this.props.className || "";
+    const className = `p-promote-button p-icon-button p-button--${appearance} ${
```

Comment:
Maybe create as an array and move the ternary's to if's for readability?

Reply:
Yeah... I hate building these long conditional class names as well. Probably worth looking into [classNames](https://github.com/JedWatson/classnames) module sooner then later.